org.palladiosimulator.analyzer.slingshot.behavior.spd.adjustment/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/monitor/ResourceContainerMonitorCloner.java

```diff
@@ -0,0 +1,167 @@
+package org.palladiosimulator.analyzer.slingshot.behavior.spd.monitor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.palladiosimulator.edp2.models.measuringpoint.MeasuringPoint;
+import org.palladiosimulator.edp2.models.measuringpoint.MeasuringPointRepository;
+import org.palladiosimulator.monitorrepository.Monitor;
+import org.palladiosimulator.monitorrepository.MonitorRepository;
+import org.palladiosimulator.monitorrepository.MonitorRepositoryFactory;
+import org.palladiosimulator.pcm.resourceenvironment.ProcessingResourceSpecification;
+import org.palladiosimulator.pcm.resourceenvironment.ResourceContainer;
+import org.palladiosimulator.pcmmeasuringpoint.ActiveResourceMeasuringPoint;
+import org.palladiosimulator.pcmmeasuringpoint.PcmmeasuringpointFactory;
+import org.palladiosimulator.pcmmeasuringpoint.ResourceContainerMeasuringPoint;
+
+/**
+ * A class that clones monitors (and measuring points) of a resource container
+ * and its active resource specifications.
+ * 
+ * Everytime a cloning action is performed, the monitor is also saved directly
+ * into the repository.
+ * 
+ * @author Julijan Katic
+ */
+public class ResourceContainerMonitorCloner {
```

Comment:
The example is actually already given in the code (line 87). The way monitors are copied is different for ResourceContainers and ActiveResourceSpecs. Since the id's of the specs are re-calculated after copying, we cannot associate the new spec with the original one anymore (i.e. from which spec is this copy?) Currently, I handle this by looking whether there is a spec of the same type (CPU, HDD). 

We need to know this, because in the scenario where there are two different active resource specs of different types in one resource container, but only for example the CPU spec has a measuring point, we must ensure that in the copy the CPU spec also has a measuring point, and the HDD spec does not. (Otherwise, we could mistakenly create one for the other spec and not for the actual spec)